src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs

```diff
@@ -294,10 +294,10 @@ await onMessage(messageContext)
                     actionToTake = await onError(errorContext)
                         .ConfigureAwait(false);
                 }
-                catch (Exception)
+                catch (Exception ex)
                 {
-                    actionToTake = await onError(errorContext)
-                        .ConfigureAwait(false);
+                    criticalError.Raise($"Failed to execute recoverability actions for message `{messageContext.MessageId}`", ex);
```

Comment:
Should it be "action" instead of "actions"?

Reply:
Yea, I copied it from sqltransport :) 

Will change it there as well

Reply:
Went with `policy` to align with https://docs.particular.net/nservicebus/recoverability/custom-recoverability-policy

Reply:
Which makes me to start to see value in asserting on the actual message to make sure all transports align?

Reply:
I've changed the tests to assert on the exact wording of the failure message